src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs

```diff
@@ -390,5 +393,43 @@ private bool IncludeDiagnostic(DiagnosticData diagnostic)
             var linePositionSpan = DiagnosticData.GetLinePositionSpan(diagnosticDataLocation, text, useMapped: true);
             return ProtocolConversions.LinePositionToRange(linePositionSpan);
         }
+
+        internal TestAccessor GetTestAccessor() => new TestAccessor(this);
+
+        internal class TestAccessor
+        {
+            private readonly InProcLanguageServer _server;
+
+            internal TestAccessor(InProcLanguageServer server)
+            {
+                _server = server;
+            }
+
+            internal ImmutableArray<Uri> GetFileUrisInPublishDiagnostics()
+                => _server._publishedFileToDiagnostics.Keys.ToImmutableArray();
+
+            internal ImmutableArray<DocumentId> GetDocumentIdsInPublishedUris()
+                => _server._documentsToPublishedUris.Keys.ToImmutableArray();
+
+            internal IImmutableSet<Uri> GetFileUrisForDocument(DocumentId documentId)
+            {
+                if (_server._documentsToPublishedUris.TryGetValue(documentId, out var uris))
```

Comment:
Didn't you add a Didn't you add a GetValueOrDefault?

Reply:
🤦‍♂